src/server/vnparse.c

```diff
@@ -3341,6 +3341,13 @@ pbs_release_nodes_given_select(relnodes_input_t *r_input, relnodes_input_select_
 		return (1);
 	}
 	err_msg[0] = '\0';
+	/* vnodes assigned to the job satisfy ATTR_SchedSelect */
+	if (r_input2->select_str &&
+		r_input->schedselect &&
+		strstr(r_input->schedselect, "vntype=cray_") != NULL) {
```

Comment:
People using crays should request vntype.  This doesn't mean people need to.  If someone doesn't request vntype, they will get what they get, but a cray reservation will be made.  This means the schedselect won't have vntype, but the job will have a cray reservation.  Since the job has a cray reservation, this check will not match and the nodes will be shrunk.